File: 05 Hands-on Activity 1 - ARG/MovieTime.py
from collections import deque

movies = deque()
snacks = deque()

for _ in range(3):
  movies.append(input("Enter a movie: "))

for _ in range(3):
  snacks.append(input("Enter a snack: "))

print("Movies:")
for movie in movies:
  print(movie)

print("Snacks:")
for snack in snacks:
  print(snack)

while True:
  choice = input("Press 'S' to finish a snack: ").upper()
  if choice == "S":
    finished_snack = snacks.popleft()
    if finished_snack:
      print(f"Finished {finished_snack}")
      print("Remaining snacks:")
      for snack in snacks:
        print(snack)
    else:
      print("No more snacks!")
      break

File: 05 Hands-on Activity 1 - ARG/MovieTime.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class MovieTime {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);

    Queue<String> movies = new LinkedList<>();
    Queue<String> snacks = new LinkedList<>();

    System.out.println("Enter three movies:");
    for (int i = 0; i < 3; i++) {
      movies.add(scanner.nextLine());
    }

    System.out.println("Enter three snacks:");
    for (int i = 0; i < 3; i++) {
      snacks.add(scanner.nextLine());
    }

    System.out.println("Movies:");
    for (String movie : movies) {
      System.out.println(movie);
    }

    System.out.println("Snacks:");
    for (String snack : snacks) {
      System.out.println(snack);
    }

    while (true) {
      System.out.println("Press 'S' to finish a snack:");
      String choice = scanner.nextLine();
      if (choice.equalsIgnoreCase("S")) {
        String finishedSnack = snacks.poll();
        if (finishedSnack != null) {
          System.out.println("Finished " + finishedSnack);
          System.out.println("Remaining snacks:");
          for (String snack : snacks) {
            System.out.println(snack);
          }
        } else {
          System.out.println("No more snacks!");
          break;
        }
      }
    }
  }
}